File: Problem-057/solution.js
function squareRootConvergents(n) {
  let count = 0;
  let num = 3n;
  let den = 2n;
  for (let i = 1; i <=n;i++){
    if (num.toString().length> den.toString().length) count++;
    [num, den] = [num+ 2n * den,num+ den];
  }
  return count;
}

File: Problem-057/solution.py
def square_root_convergents(n):

    count = 0

    numerator = 3
    denominator = 2

    for i in range(n):

        # Check if numerator has more digits
        if len(str(numerator)) > len(str(denominator)):
            count += 1

        # Update both simultaneously
        numerator, denominator = (
            numerator + 2 * denominator,
            numerator + denominator
        )

    return count


print(square_root_convergents(1000))
